### ISSUE TITLE
从0开始重新训练，训练无法收敛，val测试效果才0.64

### ISSUE THREAD

**Issue by Exception-star** [junfu1115/DANet#84]
重新训练网络，训练无法收敛，val测试效果才0.64

**Comment by VincentXWD** [comment]
Me, too. All settings are the same but got worse results:
1 road  	97.289%
2 sidewalk	79.148%
3 building	89.503%
4 wall  	47.240%
5 fence 	47.246%
6 pole  	50.839%
7 traffic light	46.385%
8 traffic sign	60.880%
9 vegetation	91.026%
10 terrain	60.514%
11 sky  	93.516%
12 person	72.177%
13 rider	44.413%
14 car  	92.095%
15 truck	54.443%
16 bus  	68.614%
17 train	63.096%
18 motorcycle	49.248%
19 bicycle	67.884%
mean_IU 	67.134%	mean_pixel_ACC	94.496%

**Comment by VincentXWD** [comment]
BTW, maybe it is caused by weight initialization. Use imagenet to initialize it will be better.

**Comment by XUYUNYUN666** [comment]
> 重新训练网络，训练无法收敛，val测试效果才0.64

你什么硬件配置呀？

**Comment by Exception-star** [comment]
> > 重新训练网络，训练无法收敛，val测试效果才0.64
> 
> 你什么硬件配置呀？
4个k40,backone:resnet50
我这个结果是训练输出的结果，并不是真实的，使用test.py会获得更好的结果

**Comment by Exception-star** [comment]
> BTW, maybe it is caused by weight initialization. Use imagenet to initialize it will be better.
训练输出的结果，并不是真实的，使用test.py会获得更好的结果

**Comment by Exception-star** [comment]
> BTW, maybe it is caused by weight initialization. Use imagenet to initialize it will be better.

Have you solved the problem of worse results？

**Comment by VincentXWD** [comment]
> > BTW, maybe it is caused by weight initialization. Use imagenet to initialize it will be better.
> 
> Have you solved the problem of worse results？

Sure, Since I trained backbone params and segmentation header params with two different learning rate, I got worse results because I forgot to set the optimizer to optimize the DA parts. Now I can get the result correctly. I got the result around 80% in multi-scale evaluating phase without bells and whistles.

**Comment by Exception-star** [comment]
> > > BTW, maybe it is caused by weight initialization. Use imagenet to initialize it will be better.
> > 
> > 
> > Have you solved the problem of worse results？
> 
> Sure, Since I trained backbone params and segmentation header params with two different learning rate, I got worse results because I forgot to set the optimizer to optimize the DA parts. Now I can get the result correctly. I got the result around 80% in multi-scale evaluating phase without bells and whistles.

How do you set up the training optimizer? Like this?
```
base_lr = 0.003
parameters_list = [{'params': model.pretrained.parameters(), 'lr': base_lr},
                              {'params': model.head.parameters(), 'lr': base_lr*10}]
optimizer = torch.optim.SGD(parameters_list, lr=base_lr, momentum=0.9, weight_decay=1e-4)
```

**Comment by VincentXWD** [comment]
> > > > BTW, maybe it is caused by weight initialization. Use imagenet to initialize it will be better.
> > > 
> > > 
> > > Have you solved the problem of worse results？
> > 
> > 
> > Sure, Since I trained backbone params and segmentation header params with two different learning rate, I got worse results because I forgot to set the optimizer to optimize the DA parts. Now I can get the result correctly. I got the result around 80% in multi-scale evaluating phase without bells and whistles.
> 
> How do you set up the training optimizer? Like this?
> 
> ```
> base_lr = 0.003
> parameters_list = [{'params': model.pretrained.parameters(), 'lr': base_lr},
>                               {'params': model.head.parameters(), 'lr': base_lr*10}]
> optimizer = torch.optim.SGD(parameters_list, lr=base_lr, momentum=0.9, weight_decay=1e-4)
> ```

Yes, my settings seem like this. I followed the instructions for setting it up as described in the paper.

**Comment by Exception-star** [comment]
> > > BTW, maybe it is caused by weight initialization. Use imagenet to initialize it will be better.
> > 
> > 
> > Have you solved the problem of worse results？
> 
> Sure, Since I trained backbone params and segmentation header params with two different learning rate, I got worse results because I forgot to set the optimizer to optimize the DA parts. Now I can get the result correctly. I got the result around 80% in multi-scale evaluating phase without bells and whistles.


I can only get 0.69（ss） result with 1 * rtx2080ti, and other training hyper params settings are the same as the paper

**Comment by XUYUNYUN666** [comment]
> 重新训练网络，训练无法收敛，val测试效果才0.64

你什么版本的pytorch

**Comment by huang229** [comment]
@VincentXWD  请问下你是什么版本的pytorch？如果方便的话能否介绍一下整体配置的环境，谢谢！！

**Comment by Yvette1993** [comment]
> > > > > BTW, maybe it is caused by weight initialization. Use imagenet to initialize it will be better.
> > > > 
> > > > 
> > > > Have you solved the problem of worse results？
> > > 
> > > 
> > > Sure, Since I trained backbone params and segmentation header params with two different learning rate, I got worse results because I forgot to set the optimizer to optimize the DA parts. Now I can get the result correctly. I got the result around 80% in multi-scale evaluating phase without bells and whistles.
> > 
> > 
> > How do you set up the training optimizer? Like this?
> > ```
> > base_lr = 0.003
> > parameters_list = [{'params': model.pretrained.parameters(), 'lr': base_lr},
> >                               {'params': model.head.parameters(), 'lr': base_lr*10}]
> > optimizer = torch.optim.SGD(parameters_list, lr=base_lr, momentum=0.9, weight_decay=1e-4)
> > ```
> 
> Yes, my settings seem like this. I followed the instructions for setting it up as described in the paper.

Excuse me. Could you please tell me your environment about the project? I meet more~~ problems. Thank you very much.
